packages/chevrotain/src/scan/lexer.ts

```diff
@@ -911,15 +878,16 @@ export function performWarningRuntimeChecks(
 ): ILexerDefinitionError[] {
   const warnings = []
   let hasAnyLineBreak = false
-  const allTokenTypes = compact(flatten(values(lexerDefinition.modes)))
+  const allTokenTypes = ([] as TokenType[])
+    .concat(...Object.values(lexerDefinition.modes ?? {}))
```

Comment:
`lexerDefinition.modes: MultiModesDefinition`, so it should not be null or undefined, but the `?? {}` is still necessary to make the test suite pass. So there might be a bad typecast somewhere.

Lodash's `values` function hides this issue, as `values(undefined)` yields `[]`, whereas `Object.values(undefined)` throws an error.